virtual_ecosystem/models/plants/plants_model.py

```diff
@@ -223,27 +248,44 @@ def _setup(
             **kwargs: Further arguments to the setup method.
         """
 
-        # Set the class attributes
+        # Set the instance attributes from the __init__ arguments
         self.flora = flora
         self.model_constants = model_constants
-        self.communities = PlantCommunities(self.data, self.flora)
+        self.communities = PlantCommunities(
+            data=self.data, flora=self.flora, grid=self.grid
+        )
+        # This is widely used internally so store it as an attribute.
+        self._canopy_layer_indices = self.layer_structure.index_canopy
 
-        # Update canopy layers
-        # TODO - this initialisation step may move somewhere else at some point.
+        # Initialise the canopy layer arrays.
+        # TODO - this initialisation step may move somewhere else at some point see #442
         self.data = initialise_canopy_layers(
             data=self.data,
             layer_structure=self.layer_structure,
         )
 
-        # This is widely used internally so store it as an attribute.
-        self._canopy_layer_indices = self.layer_structure.index_canopy
+        # Calculate the community canopy representations.
+        self.canopies = calculate_canopies(
+            communities=self.communities,
+            max_canopy_layers=self.layer_structure.n_canopy_layers,
+        )
+
+        # TODO - #697 these need to be configurable
+        self.pmodel_consts = PModelConst()
+        self.pmodel_core_consts = CoreConst()
 
-        # Run the canopy initialisation - update the canopy structure from the initial
-        # cohort data and then initialise the irradiance using the first observation for
-        # PPFD.
+        # Create and populate the canopy data layers and set the absorption from the
+        # first time index
         self.update_canopy_layers()
         self.set_canopy_absorption(time_index=0)
```

Comment:
The new abiotic model also creates some initial canopy absorption, let's check how the order works out and I might be able to delete this in the abotic model